tests/integration/security/egress_gateway_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,181 @@
+)
+
+// TestEgressGatewayTls brings up an SDS enabled cluster and will ensure that the TLS origination at
+// egress gateway allows secure communication between the egress gateway and external workload.
+// This test brings up an egress gateway to originate TLS connection. The test will ensure that requests
+// are routed securely through the egress gateway and that the TLS origination happens at the gateway.
+func TestSdsEgressGatewayTls(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			ctx.RequireOrSkip(environment.Kube)
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "egress-gateway-tls-origination",
+				Inject: true,
+			})
+			applySetupConfig(ctx, ns)
+
+			testCases := map[string]struct {
+				destinationRulePath string
+				configPath          string
+				response            string
+			}{
+				"SIMPLE TLS origination from egress gateway succeeds": {
+					destinationRulePath: simpleTLSDestinationRuleConfig,
+					configPath:          simpleTLSGatewayConfig,
+					response:            response.StatusCodeOK,
+				},
+				"No TLS origination from egress gateway returns 503 response": {
+					destinationRulePath: disableTLSDestinationRuleConfig,
+					configPath:          simpleTLSGatewayConfig,
+					response:            response.StatusCodeUnavailable,
+				},
+			}
+
+			for name, tc := range testCases {
+				ctx.NewSubTest(name).
+					Run(func(ctx framework.TestContext) {
+						doOriginationTest(ctx, ns, tc.destinationRulePath, tc.configPath, tc.response)
+					})
+			}
+		})
+}
+
+func doOriginationTest(
+	ctx framework.TestContext, ns namespace.Instance, destinationRulePath, configPath, expectedResp string) {
+	var client echo.Instance
+	echoboot.NewBuilderOrFail(ctx, ctx).
+		With(&client, util.EchoConfig("client", ns, false, nil, p)).
+		BuildOrFail(ctx)
+	ctx.ApplyConfigOrFail(ctx, ns.Name(), file.AsStringOrFail(ctx, destinationRulePath))
+	ctx.ApplyConfigOrFail(ctx, ns.Name(), file.AsStringOrFail(ctx, configPath))
+	defer ctx.DeleteConfigOrFail(ctx, ns.Name(), file.AsStringOrFail(ctx, configPath))
+
+	// give the configuration a moment to kick in
+	time.Sleep(time.Second * 20)
+	pretestReqCount := getEgressRequestCountOrFail(ctx, ns, prom)
+
+	for i := 0; i < externalReqCount; i++ {
+		w := client.WorkloadsOrFail(ctx)[0]
+		responses, err := w.ForwardEcho(context.TODO(), &epb.ForwardEchoRequest{
+			Url:   externalURL,
+			Count: 1,
+		})
+		if err != nil {
+			ctx.Fatalf("failed to make request from echo instance to %s: %v", externalURL, err)
+		}
+		if len(responses) < 1 {
+			ctx.Fatalf("received no responses from request to %s", externalURL)
+		}
+		resp := responses[0]
+
+		if expectedResp != resp.Code {
+			ctx.Errorf("expected status %s but got %s", expectedResp, resp.Code)
+		}
+
+	}
+
+	// give prometheus some time to ingest the metrics
```

Comment:
noted above, but I don't think we need this check. Is there some reason metrics will give additional coverage that the above checks do not?

Reply:
same as above. Will remove these.